src/components/CatalogLayerContextMenu.vue

```diff
@@ -32,7 +32,43 @@
         </li>
       </ul>
     </li>
-    <li v-if="canZoom(layerMenu.layer)" @click.prevent.stop="zoomToLayer(layerMenu.layer)">
+    <!--
+      @since v3.8
+    -->
+    <li
+      v-if="layerMenu.layer.geolayer && layerMenu.layer.visible"
+      @mouseleave.self="showLayerOpacityMenu(false,$event)"
+      @mouseover.self="showLayerOpacityMenu(true,$event)"
+      class="menu-icon"
+    >
+      <span class="menu-icon skin-color-dark" :class="g3wtemplate.getFontClass('slider')"></span>
+      <span class="item-text" v-t="'catalog_items.contextmenu.layer_opacity'"></span>
+      <span class="menu-icon" style="position: absolute; right: 0; margin-top: 3px" :class="g3wtemplate.getFontClass('arrow-right')"></span>
+      <ul
+        v-show="layerMenu.layer && layerMenu.layerOpacityMenu.show"
+        style="position:fixed; padding-left: 0; background-color: #FFFFFF; color:#000000"
+        :style="{
+          top: layerMenu.layerOpacityMenu.top + 'px',
+          left: `${layerMenu.layerOpacityMenu.left}px`,
+          maxHeight: layerMenu.layerOpacityMenu.maxHeight + 'px',
+          overflowY: layerMenu.layerOpacityMenu.overflowY
+        }">
+        <li>
+          <range :value="layerMenu.layer.opacity"
+            :min="0"
+            :max="100"
+            :step="1"
+            :sync="false"
+            :showValue="true"
+            :unit="'%'"
+            @changed="closeLayerMenu(layerMenu.layerOpacityMenu)"
+            @change-range="setLayerOpacity">
+          </range>
+        </li>
+      </ul>
+    </li>
+
```

Comment:
Find out how to extract this section into a new separate single file component

Reply:
@Raruto ok